src/app/recipes/store/recipe.reducer.ts

```diff
@@ -28,20 +27,19 @@ export function recipeReducer(
       };
 
     case RecipesActions.UPDATE_RECIPE:
-      const updatedRecipe = {
-        ...state.recipes[action.payload.index],
-        ...action.payload.newRecipe,
-      };
-
-      const updatedRecipes = [...state.recipes];
-      updatedRecipes[action.payload.index] = updatedRecipe;
-
+      const updatedRecipes = [...state.recipes].map((recipe) => {
+        if (recipe._id === action.payload._id) {
+          return action.payload;
+        } else {
+          return recipe;
+        }
+      });
```

Comment:
Run this logic after effect retuned success. You can create separate action for modifying your store after Back End (BE) returned success

Reply:
fixed